### PR TITLE
Respect CancellationToken by ASTVisitor

### PR DESCRIPTION
**Pull request by sungam3r**
I just forgot 🤦‍♂️ .

### CODE REVIEW

File: src/GraphQLParser.Tests/Visitors/ASTVisitorTests.cs [hunk 1]
```diff
@@ -0,0 +1,60 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using GraphQLParser.AST;
+using GraphQLParser.Visitors;
+using Shouldly;
+using Xunit;
+
+namespace GraphQLParser.Tests.Visitors;
```
Comment (sungam3r) [inline src/GraphQLParser.Tests/Visitors/ASTVisitorTests.cs:9]: It's renamed `DefaultNodeVisitorTests.cs` , I only added `ASTVisitor_Should_Respect_CancellationToken`

File: src/GraphQLParser/Visitors/ASTVisitor.cs [hunk 2]
```diff
@@ -598,6 +598,8 @@ protected virtual async ValueTask VisitInputObjectTypeExtensionAsync(GraphQLInpu
     /// <param name="context">Context passed into all INodeVisitor.VisitXXX methods.</param>
     public virtual ValueTask VisitAsync(ASTNode? node, TContext context)
     {
+        context.CancellationToken.ThrowIfCancellationRequested();
```
Comment (sungam3r) [inline src/GraphQLParser/Visitors/ASTVisitor.cs:601]: One-line fix :)

Comment (Shane32) [reply src/GraphQLParser/Visitors/ASTVisitor.cs:601]: Is this ever called recursively?

**Review by Shane32 — APPROVED** [review]
See other comments

File: src/GraphQLParser.Tests/Visitors/SDLWriterTests.cs [hunk 3]
```diff
@@ -664,7 +664,7 @@ public async Task WriteDocumentVisitor_Should_Throw_On_Unknown_Values(string tex
 
     private class Context : IASTVisitorContext
     {
-        public CancellationToken CancellationToken => throw new NotImplementedException();
+        public CancellationToken CancellationToken { get; set; }
```
Comment (Shane32) [inline src/GraphQLParser.Tests/Visitors/SDLWriterTests.cs:667]: :+1:

File: src/GraphQLParser/Visitors/ASTVisitor.cs (hunk 2 above)
Comment (sungam3r) [reply src/GraphQLParser/Visitors/ASTVisitor.cs:601]: > Is this ever called recursively?

Of course.

Comment (Shane32) [reply src/GraphQLParser/Visitors/ASTVisitor.cs:601]: Then I’d probably not add it.

Comment (Shane32) [reply src/GraphQLParser/Visitors/ASTVisitor.cs:601]: Whatever you think. I’ve never timed how slow that method is.

Comment (sungam3r) [reply src/GraphQLParser/Visitors/ASTVisitor.cs:601]: ?

Comment (sungam3r) [reply src/GraphQLParser/Visitors/ASTVisitor.cs:601]: That method dispatches call to some other method and so on.

**Pull request merged**